### ISSUE TITLE
Issue with Weekly Plotting when Number of Days of Available Data is Low

### ISSUE THREAD

**Issue by ajsockol** [ARM-DOE/ACT#187]
There seems to be an issue with weekly plots when only a small amount of data is avaialable to generate the plots with. The x-axis shows up as all zeros. This appears to correct itself once there are ~3 full available days of data, as in this example:
![srg](https://user-images.githubusercontent.com/49569746/74380879-06aff880-4db0-11ea-8ddc-02939376b658.png)

![srg2](https://user-images.githubusercontent.com/49569746/74380956-2d6e2f00-4db0-11ea-8b47-c7b89181a44f.png)

**Comment by AdamTheisen** [comment]
@ajsockol can you post the code used to create the first plot?

**Comment by ajsockol** [comment]
# Import necessary standard libraries
    from os import environ
    import datetime as dt
    import astral, act, pathlib
    from dqlib.io.data import read_arm_data_act
    from dqlib.utils.utils import DatastreamParser, string_dates
    from dqlib.io.image import save_plot

    def plot_module(run_parameters, ds_object, days=0):
    
        #Get all the file parameters and metadata needed for processing

        #Get AMC Data
        filename = run_parameters['filename']
        params = DatastreamParser(filename)
        date = params.date
        datastream = params.datastream_standard
        site = params.site
        platform = params.datastream_class
        facility = params.facility
        level = params.level
        data_path = run_parameters['base_data_path']

        #Sets default x-range to one day
        xrng = [(dt.datetime.strptime(date,'%Y%m%d') -
             dt.timedelta(days=days)),
            (dt.datetime.strptime(date,'%Y%m%d') +
             dt.timedelta(days=1))]

        #If keyword day is > 0, then weekly plots are made
        if days > 0:
            sdate, edate = tuple(string_dates(xrng, format='full'))
            ds_object = read_arm_data_act(
                datastream, startdate=sdate, enddate=edate,
                basepath=data_path)
            plotname = 'SRG_weekly'
        else:
            plotname = 'SRG_daily'

        # ----------------------------------
        # Plot data
        # ----------------------------------
    
        #Make SRG Plots

        #Create figure and set size by multiplying figheight
        #by the number of figures to plot from subplot_shape
        figwidth = 10  # width of figure in pixels. 9 = 900 pixels
        figheight = 3  # height of figure in pixels. 3 = 300 pixels
        subplot_shape = (3, ) #Number of plots in figure

        #Use ACT time series display
        display = act.plotting.TimeSeriesDisplay(
            {datastream:ds_object}, subplot_shape=subplot_shape,
            figsize=(figwidth, subplot_shape[0]*figheight))
        display.fig.tight_layout()
        display.fig.subplots_adjust(left=0.1,right=0.85,bottom=0.08,
            top=0.95,hspace=0.3)

        var_names = ['battery_voltage_min','panel_temperature','siphon_rain_gauge_adc']
        subplot_titles = ['Minimum Battery Voltage','Panel Temperature',
            'Instantaneous Precipitation Amount of Siphon Gauge']
        subplot_indices = [0,1,2]

        for ii, var in enumerate(var_names):
            subplot_index = subplot_indices[ii]
            display.plot(var, subplot_index=(subplot_index, ), label=var)
            leg2 = display.axes[subplot_index].legend(loc='center left', bbox_to_anchor=(0.99, 0.5),prop={'size':8},
            frameon=False, markerscale=0, handlelength=0)
        for line, text in zip(leg2.get_lines(), leg2.get_texts()):
            text.set_color(line.get_color())
        display.axes[subplot_index].autoscale(tight=False,axis='y')
        display.axes[subplot_index].set_xlim(xrng)
        if days > 0:
            display.axes[subplot_index].set_title(datastream + ' ' + subplot_titles[subplot_index] + ' from ' + sdate + ' to ' + date)
        else:
            display.axes[subplot_index].set_title(datastream + ' ' + subplot_titles[subplot_index] + ' on ' + date)

        try:
            display.day_night_background(datastream, subplot_index=(subplot_index, ))
        except (AttributeError, ValueError, astral.AstralError):
            pass

        #Create and save plots as png files
        imagename = '.'.join([datastream,plotname,date,'png'])
        imagename = pathlib.Path(environ['DQPLOT_DATA'], site, site+platform,
            date, imagename)
        save_plot(display.fig, imagename=imagename, run_parameters=run_parameters)

**Comment by AdamTheisen** [comment]
@ajsockol I think I figured it out.  The data are first plotting for 1-2 days, so the x-format is based on those 2 days.  You then adjust the xrng after the plotting and don't adjust the format.  2 options

1. add the following
        myFmt = common.get_date_format(days)
        display.axes[subplot_index].xaxis.set_major_formatter(myFmt)

2. Pass your xrng to the plot call using time_rng that was added at some point.

    display.plot('battery_voltage_min', time_rng=xrng)

Option number 2 is much simpler it looks like.
